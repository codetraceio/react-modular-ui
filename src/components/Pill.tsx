import React, { PropsWithChildren, useContext } from "react";
import { className } from "../utils/className";
import { ThemeContext } from "./ThemeContext";

export interface PillProps {
  size?: string | number;
  color?: string;
  variant?: string;
  padding?: string | number;
  width?: string | number;
  theme?: string;

  onClick?: () => void;

  // deprecated
  paddingLeft?: string | number;
  paddingRight?: string | number;
}

export default function Pill({
  size,
  color,
  variant,
  paddingLeft,
  paddingRight,
  padding,
  width,
  theme,
  children,
  onClick,
  ...props
}: PropsWithChildren<PillProps>) {
  const themeContext = useContext(ThemeContext);

  return (
    <span
      className={className("pill")}
      data-size={size}
      data-color={color}
      data-variant={variant}
      data-padding-left={paddingLeft}
      data-padding-right={paddingRight}
      data-padding={padding}
      data-width={width}
      data-theme={theme ?? themeContext}
      onClick={onClick}
      {...props}
    >
      {children}
    </span>
  );
}
