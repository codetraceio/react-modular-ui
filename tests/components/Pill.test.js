import * as React from "react";
import * as renderer from "react-test-renderer";

import Pill from "../../src/components/Pill";

describe("index", () => {
  it("should render Pill", () => {
    const tree = renderer.create(
      <Pill
        size="32"
        type="primary"
      >
        Something
      </Pill>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
