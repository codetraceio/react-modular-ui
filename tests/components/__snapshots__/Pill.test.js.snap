// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index should render Pill 1`] = `
<div
  className="c-pill -size-32"
>
  Something
</div>
`;
